### PR TITLE
Add global phase to `MottonenStatePreparation`

### PR DESCRIPTION
**Pull request by soranjh**
**Context:**
Fixes this [bug](https://github.com/PennyLaneAI/pennylane/issues/5099) by adding a global phase to `qml.MottonenStatePreparation`.

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

**Comment by soranjh** [comment]
[sc-55294]

File: tests/templates/test_state_preparations/test_mottonen_state_prep.py [hunk 1]
```diff
@@ -133,27 +133,17 @@ class TestDecomposition:
...
-        results = circuit()
+        state = circuit()
 
-        state = results[-1].ravel()
-        fidelity = abs(np.vdot(state, target_state)) ** 2
-
-        # We test for fidelity here, because the vector themselves will hardly match
-        # due to imperfect state preparation
-        assert np.isclose(fidelity, 1, atol=tol, rtol=0)
+        assert np.allclose(state, target_state)
```
Comment (soranjh) [inline tests/templates/test_state_preparations/test_mottonen_state_prep.py:146]: The fidelity test was probably added to account for the phase mismatch. Now, we can directly compare the computed and target states.

**Review by timmysilv — APPROVED** [review]
don't forget the changelog entry! looks great otherwise

**Review by albi3ro — APPROVED** [review]

**Comment by albi3ro** [comment]
Screenshot 2024-02-07 at 3 22 43 PM

**Pull request merged**